### PR TITLE
chore: update pytest

### PR DESCRIPTION
**Pull request by pauldambra**
pytest isn't up to date... what will people think?!

(pip installing our version of pytest was showing a warning about future lack of support so we may as well update)

All the tests passed so 🤷

### CODE REVIEW

File: posthog/api/test/__snapshots__/test_feature_flag.ambr [hunk 1]
```diff
@@ -332,25 +333,10 @@
      WHERE team_id = 2
      GROUP BY id
      HAVING max(is_deleted) = 0)
-  '
----
-# name: TestResiliency.test_feature_flags_v3_with_a_working_slow_db
```
Comment (daibhin) [inline posthog/api/test/__snapshots__/test_feature_flag.ambr:337]: Here

File: posthog/api/test/__snapshots__/test_insight.ambr [hunk 2]
```diff
@@ -1469,28 +1470,10 @@
   WHERE (NOT ("posthog_dashboard"."deleted")
          AND "posthog_dashboard"."id" = 2)
   LIMIT 21 /*controller='project_insights-list',route='api/projects/%28%3FP%3Cparent_lookup_team_id%3E%5B%5E/.%5D%2B%29/insights/%3F%24'*/
-  '
----
-# name: TestInsight.test_listing_insights_does_not_nplus1.30
```
Comment (daibhin) [inline posthog/api/test/__snapshots__/test_insight.ambr:1474]: Here

File: posthog/api/test/__snapshots__/test_element.ambr [hunk 3]
```diff
@@ -103,39 +104,27 @@
...
   SELECT "posthog_instancesetting"."id",
          "posthog_instancesetting"."key",
          "posthog_instancesetting"."raw_value"
   FROM "posthog_instancesetting"
   WHERE "posthog_instancesetting"."key" = 'constance:posthog:HEATMAP_SAMPLE_N'
   ORDER BY "posthog_instancesetting"."id" ASC
   LIMIT 1 /*controller='element-stats',route='api/element/stats/%3F%24'*/
-  '
----
-# name: TestElement.test_element_stats_postgres_queries_are_as_expected.5
```
Comment (daibhin) [inline posthog/api/test/__snapshots__/test_element.ambr:130]: Here

File: ee/clickhouse/queries/test/__snapshots__/test_retention.ambr [hunk 4]
```diff
@@ -952,221 +743,5 @@
   ORDER BY length(appearances) DESC, actor_id
   LIMIT 100
   OFFSET 0
-  '
----
-# name: TestClickhouseRetention.test_retention_event_action
```
Comment (daibhin) [inline ee/clickhouse/queries/test/__snapshots__/test_retention.ambr:957]: Here

File: ee/clickhouse/queries/test/__snapshots__/test_retention.ambr [hunk 5]
```diff
@@ -744,96 +621,10 @@
            intervals_from_base
   ORDER BY breakdown_values,
            intervals_from_base
-  '
----
-# name: TestClickhouseRetention.test_groups_filtering_person_on_events_v2.2
```
Comment (daibhin) [inline ee/clickhouse/queries/test/__snapshots__/test_retention.ambr:749]: Here

File: ee/clickhouse/queries/test/__snapshots__/test_retention.ambr [hunk 6]
```diff
@@ -518,134 +519,10 @@
            intervals_from_base
   ORDER BY breakdown_values,
            intervals_from_base
-  '
----
-# name: TestClickhouseRetention.test_groups_filtering_person_on_events_materialized
```
Comment (daibhin) [inline ee/clickhouse/queries/test/__snapshots__/test_retention.ambr:523]: Here

File: ee/api/test/__snapshots__/test_organization_resource_access.ambr [hunk 7]
```diff
@@ -115,24 +116,10 @@
   FROM "ee_organizationresourceaccess"
   WHERE "ee_organizationresourceaccess"."organization_id" = '00000000-0000-0000-0000-000000000000'::uuid
   LIMIT 100 /*controller='organization_resource_access-list',route='api/organizations/%28%3FP%3Cparent_lookup_organization_id%3E%5B%5E/.%5D%2B%29/resource_access/%3F%24'*/
-  '
----
-# name: TestOrganizationResourceAccessAPI.test_list_organization_resource_access_is_not_nplus1.15
```
Comment (daibhin) [inline ee/api/test/__snapshots__/test_organization_resource_access.ambr:120]: Here

**Review by daibhin — APPROVED** [review]
Did a pass through. Most of the changes seem inconsequential but just marked anywhere where a query was removed - just marked all of them I came across with `Here` (possibly missed a few). Didn't spend any time looking into why but I assume they're not needed given the test themselves haven't changed.

The actual change seems good :)

**Comment by pauldambra** [comment]
wat

I'm going to start from scratch cos I can't figure out why those snapshots are deleted :'(

**Comment by neilkakkar** [comment]
oh, don't get discouraged, I think (low context) that this is doing the right thing. Here's the 'bug':

1. Snapshots are sometimes left over, if they are at the end (ex: new function has 8 queries, but previously it had 10), so the last 2 would randomly stick around.
2. Same when the function is deleted/renamed, snapshots sometimes aren't.
3. snapshot decorator was deleted at a later stage, old snapshots weren't cleaned up

From a glance, these bugs seem to be fixed in the new version, since the deletions fall in here.

So, I'd say, thank you for the good work, let's get this in? xD

**Comment by neilkakkar** [comment]
(although if you run it locally and see the snapshot but see it fail here, ignore me, something is indeed wrong then 😬 )

**Pull request merged**